products/accredited-representation-management/technical-documentation/appoint-a-representative/poa_request_proposal.md

```diff
@@ -0,0 +1,160 @@
+# Storage Proposal for Power of Attorney Requests
+This proposal only covers the storing of data for Power of Attorney Requests.
+
+## Background
+The Office of the Chief Technology Officer (OCTO) is seeking to migrate Veteran and Power of Attorney (PoA) experiences from the legacy Stakeholder Enterprise Platform (SEP) product and eBenefits platform to VA.gov.
+
+One aspect of this experience is the Power of Attorney (POA) Request, a data object that exists to enable accredited representatives accepting or denying requests to act as a Power of Attorney (Forms 21-22 and 21-22a) on behalf of Veterans. The data object bridges the gap from the Veteran facing form fill experience (Appoint a Representative) to establishing the Power of Attorney relationship in systems of record and updating representative permissions in those systems (Lighthouse auto-establishment APIs). [See the flow chart](#data-flow)
+
+## Problem
+In order to create a functional user interface for accredited representatives ([see the sample UI](#sample-ui)), the POA Request objects must support reasonably efficient searching or filtering on the following Veteran attributes: first name, last name, city, state, zip code, and consent limits. The front end must also be able to display information related to change of address authorization and health record access limitations.
+
+There is not an existing pattern or solution in vets-api for storing Veteran PII that is secure and efficiently searchable for a many row result.
+
+## Solution
+Code below is for illustrative purposes. These snippets are not likely to be the final implementation.
+
+### Column level encryption
+The [blind index gem](https://github.com/ankane/blind_index) allows for exact matching queries on encrypted columns. This will be used on fields that users will search or filter by. Existing KMS encryption will be used on the fields that are for display purposes only.
+
+#### The Model
+```ruby
+class PowerOfAttorneyRequest < ApplicationRecord
+  has_kms_key
+  has_encrypted :consent_data, key: :kms_key, **lockbox_options
+  blind_index :first_name, :last_name, :city, :state_code, :zip_code
+
+  belongs_to :user_account
+  belongs_to :requested_individual, class_name: 'AccreditedIndividual'
+  belongs_to :accepted_or_declined_by, class_name: 'AccreditedIndividual', optional: true
+
+  enum status: {
+    'pending' => 'pending',
+    'accepted' => 'accepted',
+    'declined' => 'declined',
+    'expired' => 'expired'
+  }
+
+  def accept!(portal_user)
+    terminate
+    self.status = 'accepted'
+    self.accepted_or_declined_by = portal_user
+
+    self.save
+  end
+
+  def decline!(portal_user, reason)
+    terminate
+    self.status = 'declined'
+    self.accepted_or_declined_by = portal_user
+    self.decline_reason = reason
```

Comment:
I suspect that this could easily have sensitive information. Particularly if there is a free-form input option.